mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxExtensions.kt

```diff
@@ -117,36 +126,19 @@ inline fun <T, reified VM : BaseMvRxViewModel<S>, reified S : MvRxState> T.activ
 @RestrictTo(LIBRARY)
 fun <T : Fragment> T._fragmentArgsProvider(): Any? = arguments?.get(MvRx.KEY_ARG)
 
-/**
- * For internal use only. Public for inline.
- *
- * Looks for [MvRx.KEY_ARG] on the arguments of the fragment receiver.
- *
- * Also adds the fragment's MvRx args to the host Activity's [MvRxViewModelStore] so that they can be used to recreate initial state
- * in a new process.
- */
-@Suppress("FunctionName")
-@RestrictTo(LIBRARY)
-inline fun <T : Fragment> T._activityArgsProvider(keyFactory: () -> String): Any? {
```

Comment:
I'm wondering about removing this now, are you sure that the functionality is made up somewhere else? I think we may be missing something.

Below you have 
```
FragmentViewModelContext -> viewModelContext.copy(args = restoredArgs)
```

which never happened before - the restored arguments were only used in activity context. 

The point of this activity args approach is to identify the fragment that created an "activityViewModel", save those arguments, and if another fragment restores that view model it can use the correct arguments from a different fragment.

I don't think this current approach lets us do that. The arguments from the first fragment get overridden, since when saving to the bundle we use `viewModelContext.args` of the last fragment to access that viewmodel and register the saved state registry.

A potential fix might be to add an internal property on BaseMvRxViewModel that stores the arguments of the fragment used to create it. Then in `getSavedStateBundle` we would save those arguments instead of the args from the context

In our app we should be able to test if this works or not by going through a flow to a page that has an "existingViewModel", where the viewmodel was initialized by arguments from the first screen. Then use debug settings to force the process to be killed when you switch apps (set it to keep 0 processes). Coming back to the app should restore the flow correctly, but right now I believe it would crash